File: App/Resources/AlgorithmDetails/lessbogosort/lessbogosort.py
import random


def is_minimum(arr, start, end):
    for k in range(start + 1, end):
        if arr[start] > arr[k]:
            return False
    return True


def shuffle_range(arr, start, end):
    for i in range(start, end - 1):
        j = random.randrange(i, end)
        arr[i], arr[j] = arr[j], arr[i]


def sort(arr):
    n = len(arr)
    for i in range(n):
        while not is_minimum(arr, i, n):
            shuffle_range(arr, i, n)


if __name__ == "__main__":
    array = [0, 39, 21, 62, 91, 77, 14, 23]
    sort(array)
    print(array)

File: App/Resources/AlgorithmDetails/lessbogosort/lessbogosort.swift
func isMinimum(_ arr: [Int], _ start: Int, _ end: Int) -> Bool {
    for k in (start + 1) ..< end {
        if arr[start] > arr[k] {
            return false
        }
    }
    return true
}

func shuffleRange(_ arr: inout [Int], _ start: Int, _ end: Int) {
    for i in start ..< (end - 1) {
        let j = Int.random(in: i ..< end)
        arr.swapAt(i, j)
    }
}

func sort(_ arr: inout [Int]) {
    let n = arr.count
    for i in 0 ..< n {
        while !isMinimum(arr, i, n) {
            shuffleRange(&arr, i, n)
        }
    }
}

var array: [Int] = [0, 39, 21, 62, 91, 77, 14, 23]
sort(&array)
print(array)
